sync-core/src/main/java/com/cloudant/sync/query/QueryValidator.java

```diff
@@ -457,18 +471,12 @@ private static boolean validateExistsArgument(Object exists) {
         return valid;
     }
 
-    private static boolean validateQueryValue(Object value) {
+    private static boolean validateNotAFloat(Object value) {
         boolean valid = true;
-        if (value instanceof Map) {
-            for (Object key: ((Map) value).keySet()) {
-                valid = valid && validateQueryValue(((Map) value).get(key));
-            }
-        } else if (value instanceof List) {
-            for (Object element : (List) value) {
-                valid = valid && validateQueryValue(element);
-            }
-        } else if (value instanceof Float) {
+
+        if (value instanceof Float) {
             valid = false;
+            logger.log(Level.SEVERE, "Float value encountered in query");
```

Comment:
Added additional commentary to message.  Done in 5b5040e6a64ad6720ef17ace362f2020397d0880.